### PR TITLE
split existing c169 pergame hack into component parts to make research on speed racer easier

### PR DESCRIPTION
**Pull request by DavidHaywood**
the old code was hiding 3 different hacks behind a single 'm_is_namcofl' variable.

if I'm trying to phase these out individually, it makes sense to split that up

### CODE REVIEW

File: src/mame/video/namco_c169roz.cpp [hunk 1]
```diff
@@ -236,7 +285,15 @@ void namco_c169roz_device::draw(screen_device &screen, bitmap_ind16 &bitmap, con
 		if ((attrs & 0x8000) == 0)
 		{
 			// second ROZ layer is configured to use per-scanline registers
-			if (which == special && mode == 0x8000)
+
+			// Mach Breakers sets mode 0x8000 on the tilemap used by the first attract race
```
Comment (angelosa) [inline src/mame/video/namco_c169roz.cpp:289]: “Attract race” but Mach Breakers isn’t a racing tho?

File: src/mame/video/namco_c169roz.cpp [hunk 2]
```diff
@@ -97,21 +99,70 @@ void namco_c169roz_device::unpack_params(const uint16_t *source, roz_parameters
...
+
+	These tables are at
+	0x70e000  on NamcoNB1 (68k based)
+	0x30c0e000 on NamcoFL (i960 based)
+
+	the format seems to be the same as
+	m_control when not in line mode
+
+	is it possible the first line is taken from m_control?
+	as Final Lap R populates the final line as 'disbled'
```
Comment (angelosa) [inline src/mame/video/namco_c169roz.cpp:113]: _Disbled_

File: src/mame/video/namco_c169roz.cpp [hunk 3]
```diff
@@ -151,8 +202,7 @@ void namco_c169roz_device::unpack_params(const uint16_t *source, roz_parameters
 
 void namco_c169roz_device::draw_helper(screen_device &screen, bitmap_ind16 &bitmap, tilemap_t &tmap, const rectangle &clip, const roz_parameters &params)
 {
-	if (!m_is_namcofl)
-//  if (m_gametype != NAMCOFL_FINAL_LAP_R) // Fix speedrcr some title animations, but broke at road scene
+	if (!m_disable_custom_draw)
```
Comment (angelosa) [inline src/mame/video/namco_c169roz.cpp:205]: Not sure I like the definition of “custom draw”, what exactly is supposed to be?

**Comment by DavidHaywood** [comment]
I give up with this anyway, I think the data in RAM is wrong.

**Comment by DavidHaywood** [comment]
after Ryan has added the zoom to the tilemap viewer I'm starting to think the RAM content is correct, and there are just some odd modes or maybe hardcoded per board behaviors, we need to handle.

**Pull request closed without merging**